crates/revmc-worker/src/external.rs

```diff
@@ -1,51 +1,70 @@
-use std::sync::{Arc, RwLock};
+use std::{
+    fmt::{Debug, Display},
+    num::NonZeroUsize,
+    sync::{Arc, RwLock},
+};
 
 use crate::{
     error::ExtError,
     worker::{aot_store_path, CompileWorker, SledDB},
 };
 use alloy_primitives::B256;
+use lru::LruCache;
 use once_cell::sync::OnceCell;
-use revm_primitives::SpecId;
+use revm::Database;
+use revm_primitives::{AccessListItem, SpecId};
 use revmc::EvmCompilerFn;
 
 pub(crate) static SLED_DB: OnceCell<Arc<RwLock<SledDB<B256>>>> = OnceCell::new();
 
 #[derive(Debug)]
-pub struct EXTCompileWorker {
+pub struct EXTCompileWorker<DB> {
     compile_worker: Box<CompileWorker>,
+    pub cache: LruCache<String, (EvmCompilerFn, libloading::Library)>,
+    _marker: std::marker::PhantomData<DB>,
 }
 
-impl EXTCompileWorker {
-    pub fn new(threshold: u64, max_concurrent_tasks: usize) -> Self {
+impl<DB> EXTCompileWorker<DB> {
```

Comment:
Can we make where (trait bound) of DB in here? instead of `cache_load_access_list`
```
impl<DB> EXTCompileWorker<DB>
where
    DB: Database,
    <DB as revm::Database>::Error: Debug + Display,
...
```